woocommerce/woocommerce-ios
# Shipping Labels: New package details validation

Part of #4599 
⚠️ This PR depends on #4962 so please make sure that one is reviewed and merged first! ⚠️ 

# Description
This PR adds validation for total weight and handles confirm of package selection in the new Package Details screen (with multiple packages support). Validation error message is also added to make it clearer when the Done button is disabled - this is a new addition compared to the old Package Details screen.

# Changes
- Package item view model: 
   - Adds validation for total weight.
   - Returns validated package attributes if the total weight is valid.
- Package item view:
   - Shows validation error row when total weight is invalid.
- ShippingLabelPackageNumberRow:
   - Adds new error icon in case validation fails for the package. This is hidden by default to not show up in the old Package Details screen.
- Packages form view model:
   - Observes item view models for validation states and save them locally.
   - Observes validation states to check whether Done button should be enabled.
   - Adds new method to handle package selection confirm.
- Packages form: 
   - Update Done button state and trigger confirm when the button is tapped.
- Shipping Label purchase form: 
   - Updates selected packages when receiving completion from the packages form.
- Update unit tests for package item and packages form view models.

# Demo
![package-validation](https://user-images.githubusercontent.com/5533851/132820023-01aa3876-08ef-47a3-88ef-120afea4734b.gif)

ℹ️  The footer of the package item and validation error look weird when stacking them together, so I'm showing either of them based on the validation state. @woocommerce/mobile-designers - let me know what you think about this.

# Testing 
1. Make sure that your test store has installed WCShip plugin and configured packages in WPAdmin > WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Open an order detail that is eligible for the shipping label.
3. Configure Ship To and Ship From.
4. Tap Continue on Package Details row. Edit the weight to an invalid value, notice that a validation error shows up and Done button is disabled.
5. Update the weight to a valid value and tap Done. Notice on the purchase form that the correct package name and weight is displayed. Tap on Package Details row again, notice that correct package details are displayed.
6. Turn off multi-package feature flag and repeat steps 2, 3, 4, 5. Verify that everything still works as expected.

Update release notes:
